.github/workflows/pytest.yml

```diff
@@ -1,59 +1,55 @@
 name: pytest
 
 on:
-    push:
+  push:
 
 jobs:
-    pytest:
-        runs-on: blacksmith-8vcpu-ubuntu-2204
-        steps:
-        - name: Checkout code
-          uses: actions/checkout@v4
-
-        # Required due to a bug in the checkout action
-        # https://github.com/actions/checkout/issues/1471
-        - run: git fetch --prune --unshallow --tags
-
-        - name: Install uv
-          uses: useblacksmith/setup-uv@v4
-          with:
-            version: 0.5.5
-            enable-cache: true
-            cache-dependency-glob: "uv.lock"
-
-        - name: "Set up Python"
-          uses: useblacksmith/setup-python@v6
-          with:
-            python-version-file: "pyproject.toml"
-
-        - name: Setup cmake
-          uses: jwlawson/actions-setup-cmake@v2
-
-        - name: Run pytest
-          id: pytest
-          continue-on-error: true
-          run: |
-            if [[ "${{ github.ref }}" == "refs/heads/main" ]]; then
-              uv run --dev --no-editable --frozen pytest -m "not not_in_ci"
-            else
-              uv run --dev --no-editable --frozen pytest -m "not not_in_ci and not slow"
-            fi
-
-        - name: Upload all test logs
-          if: steps.pytest.outcome == 'failure'
-          uses: actions/upload-artifact@v4
-          with:
-            name: test-logs
-            path: artifacts/**/*.log
-            if-no-files-found: warn
-
-        - name: Upload all test report
-          if: steps.pytest.outcome == 'failure'
-          uses: actions/upload-artifact@v4
-          with:
-            name: test-report
-            path: artifacts/test-report.html
-
-        - name: Check pytest status
-          if: steps.pytest.outcome == 'failure'
-          run: exit 1
+  pytest:
+    runs-on: blacksmith-8vcpu-ubuntu-2204
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    # Required due to a bug in the checkout action
+    # https://github.com/actions/checkout/issues/1471
+    - run: git fetch --prune --unshallow --tags
+
+    - name: Install uv
+      uses: useblacksmith/setup-uv@v4
+      with:
+        version: 0.5.5
+        enable-cache: true
+        cache-dependency-glob: "uv.lock"
+
+    - name: "Set up Python"
+      uses: useblacksmith/setup-python@v6
+      with:
+        python-version-file: "pyproject.toml"
+
+    - name: Setup cmake
+      uses: jwlawson/actions-setup-cmake@v2
+
+    - name: Run pytest
+      id: pytest
+      continue-on-error: true
+      run: |
+        uv run --dev --no-editable --frozen pytest -m "not not_in_ci and not slow and not regression" --timeout=300
```

Comment:
Primarily, this is just a dedent / format. This line is the only one which matters

++ "not regression"
++ "timeout"